--- ../../tmp/RT-Thread_1.2.0/bsp/stm32f40x/applications/application.c	2014-01-06 22:38:34.000000000 +0800
+++ application.c	2014-02-09 23:31:36.300974507 +0800
@@ -63,62 +63,43 @@
 ALIGN(RT_ALIGN_SIZE)
 static char thread_led1_stack[1024];
 struct rt_thread thread_led1;
-static void rt_thread_entry_led1(void* parameter)
+ static void rt_thread_entry_led1(void* parameter)
 {
     GPIO_InitTypeDef  GPIO_InitStructure;
 
     /* GPIOD Periph clock enable */
-    RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_GPIOD, ENABLE);
+    RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_GPIOG, ENABLE);
 
-    /* Configure PD12, PD13, PD14 and PD15 in output pushpull mode */
-    GPIO_InitStructure.GPIO_Pin = GPIO_Pin_12 | GPIO_Pin_13| GPIO_Pin_14| GPIO_Pin_15;
+    /* Configure PD13 and PD14 in output pushpull mode */
+    GPIO_InitStructure.GPIO_Pin = GPIO_Pin_13| GPIO_Pin_14;
     GPIO_InitStructure.GPIO_Mode = GPIO_Mode_OUT;
     GPIO_InitStructure.GPIO_OType = GPIO_OType_PP;
     GPIO_InitStructure.GPIO_Speed = GPIO_Speed_100MHz;
     GPIO_InitStructure.GPIO_PuPd = GPIO_PuPd_NOPULL;
-    GPIO_Init(GPIOD, &GPIO_InitStructure);
+    GPIO_Init(GPIOG, &GPIO_InitStructure);
 
     f_var1 = 3.141592653;
     f_var2 = 123.456;
     f_var3 = 0.123456789;
     f_var4 = 1.00001;
-
     while (1)
     {
-        /* PD12 to be toggled */
-        GPIO_SetBits(GPIOD, GPIO_Pin_12);
-
-        /* Insert delay */
-        rt_thread_delay(RT_TICK_PER_SECOND/2);
-        f_var3 += f_var4;
-        f_var4 = f_var4 * f_var4;
-
         /* PD13 to be toggled */
-        GPIO_SetBits(GPIOD, GPIO_Pin_13);
-
+        GPIO_SetBits(GPIOG, GPIO_Pin_13);
         /* Insert delay */
         rt_thread_delay(RT_TICK_PER_SECOND/2);
         f_var3 += f_var4;
         f_var4 = f_var4 * f_var4;
 
         /* PD14 to be toggled */
-        GPIO_SetBits(GPIOD, GPIO_Pin_14);
-
-        /* Insert delay */
-        rt_thread_delay(RT_TICK_PER_SECOND/2);
-        f_var3 += f_var4;
-        f_var4 = f_var4 * f_var4;
-
-        /* PD15 to be toggled */
-        GPIO_SetBits(GPIOD, GPIO_Pin_15);
+//        GPIO_SetBits(GPIOG, GPIO_Pin_14);
 
         /* Insert delay */
-        rt_thread_delay(RT_TICK_PER_SECOND*2);
-        f_var3 += f_var4;
-        f_var4 = f_var4 * f_var4;
-
-        GPIO_ResetBits(GPIOD, GPIO_Pin_12|GPIO_Pin_13|GPIO_Pin_14|GPIO_Pin_15);
+//        rt_thread_delay(RT_TICK_PER_SECOND/2);
+//        f_var3 += f_var4;
+//        f_var4 = f_var4 * f_var4;
 
+        GPIO_ResetBits(GPIOG, GPIO_Pin_13|GPIO_Pin_14);
         /* Insert delay */
         rt_thread_delay(RT_TICK_PER_SECOND);
         f_var3 += f_var4;
@@ -126,7 +107,6 @@
     }
 }
 
-
 ALIGN(RT_ALIGN_SIZE)
 static char thread_led2_stack[1024];
 struct rt_thread thread_led2;
