--- ../../tmp/RT-Thread_1.2.0/bsp/stm32f40x/Libraries/SConscript	2014-01-06 22:38:34.000000000 +0800
+++ SConscript	2014-02-09 01:14:58.314117472 +0800
@@ -7,7 +7,7 @@
 
 # The set of source files associated with this SConscript file.
 src = Split("""
-CMSIS/ST/STM32F4xx/Source/Templates/system_stm32f4xx.c
+CMSIS/Device/ST/STM32F4xx/Source/Templates/system_stm32f4xx.c
 """)
 
 src += Glob('STM32F4xx_StdPeriph_Driver/src/*.c')
@@ -25,7 +25,7 @@
 
 #add for startup script 
 if rtconfig.CROSS_TOOL == 'gcc':
-     src = src + ['CMSIS/ST/STM32F4xx/Source/Templates/gcc_ride7/startup_stm32f4xx.s']
+     src = src + ['../startup_stm32f429_439xx.S']
 elif rtconfig.CROSS_TOOL == 'keil':
      src = src + ['CMSIS/ST/STM32F4xx/Source/Templates/arm/startup_stm32f4xx.s']
 # elif rtconfig.CROSS_TOOL == 'iar':
@@ -34,7 +34,7 @@
 path = [cwd + '/STM32F4xx_StdPeriph_Driver/inc', 
     cwd + '/CMSIS/ST/STM32F4xx/Include',
     cwd + '/CMSIS/Include',
-    cwd + '/CMSIS/CM3/DeviceSupport/ST/STM32F10x']
+    cwd + '/CMSIS/Device/ST/STM32F4xx/Include/']
 
 #CPPDEFINES = ['USE_STDPERIPH_DRIVER', rtconfig.STM32_TYPE]
 CPPDEFINES = ['USE_STDPERIPH_DRIVER']
